### PR TITLE
use distroless/static as base image

### PR DESCRIPTION
**Pull request by sozercan**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
- Current base image (`debian-base:v1.0.0`) has quite a few vulnerabilities, moving to `distroless/static` for better security
- We were building both ccm and cnm for each image (with `make all`), building ccm or cnm now

**Release note**:
```
Use distroless/static as base image for CCM/CNM for Linux
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,11 +12,11 @@
...
+FROM golang:1.13.9-stretch AS builder
 WORKDIR /go/src/sigs.k8s.io/cloud-provider-azure
 COPY . .
-RUN make ENABLE_GIT_COMMAND=true
+RUN make bin/azure-cloud-controller-manager ENABLE_GIT_COMMAND=true
 
-FROM k8s.gcr.io/debian-base:v1.0.0
-COPY --from=build /go/src/sigs.k8s.io/cloud-provider-azure/bin/azure-cloud-controller-manager /usr/local/bin
-RUN ln -s /usr/local/bin/azure-cloud-controller-manager /usr/local/bin/cloud-controller-manager
+FROM gcr.io/distroless/static@sha256:c6d5981545ce1406d33e61434c61e9452dad93ecd8397c41e89036ef977a88f4
```
Comment (feiskyer) [inline Dockerfile:20]: where did the sha get here? could we use a tag instead?

File: Dockerfile [hunk 2]
```diff
@@ -12,11 +12,11 @@
...
+COPY --from=builder /go/src/sigs.k8s.io/cloud-provider-azure/bin/azure-cloud-controller-manager /bin/cloud-controller-manager
```
Comment (feiskyer) [inline Dockerfile:21]: could you keep the path same?

File: Dockerfile (hunk 1 above)
Comment (sozercan) [reply Dockerfile:20]: Distroless images doesn't have semantic version tags, and we shouldn't use latest as a best practice to have deterministic builds.

File: Dockerfile (hunk 2 above)
Comment (sozercan) [reply Dockerfile]: Path is just arbitrary here since base distroless image doesn't have a shell. I changed it to `/usr/local/bin`

**Pull request merged**